### PR TITLE
Specify available track types in `showAvailableTrackPositions`

### PR DESCRIPTION
**Pull request by pkerpedjiev**
## Description

What was changed in this pull request?

When one drags a dataset over HiGlass, region outlines will pop up showing where that dataset can be displayed as a track. This relies on a JSON structure passed into `api.showAvailableTrackPositions`. Previously, we would look at the datatype in that JSON structure and determine which track types were available. Now we add the extra ability to specify which track we would expect to be created and where.

Example input JSON to `api.showAvailableTrackPositions`:

```
        {
          server: 'http://higlass.io/api/v1',
          tilesetUid: 'WtBJUYawQzS9M2WVIIHnlA',
          datatype: 'linear-labels',
          defaultTracks: ['linear-labels'],
        }
```
Also accidentally lumped in with this PR are some housekeeping files containing example commands for building the docs. Those are convenience documentation files and will likely be made obsolete in the very near future.

## Screenshot

image

## Checklist

- [x] Unit tests added or updated
- [x] Documentation added or updated
- [x] Example added or updated
- [x] Screenshot for visual changes (e.g. new tracks)
- [x] Updated CHANGELOG.md

### CODE REVIEW

File: app/scripts/api.js [hunk 1]
```diff
@@ -278,9 +278,26 @@ const createApi = function api(context, pubSub) {
...
+       * Show overlays where this track can be positioned. This
+       * function will take a track definition and display red
+       * or green overlays highlighting where the track can be
+       * placed on the view.
        *
        * @param {obj} track { server, tilesetUid, datatype }
+       *
+       * @example
+       *
+       *  var lineTrack = {
```
Comment (flekschas) [inline app/scripts/api.js]:
```suggestion
       *  let lineTrack = {
```

Let's use ES6 code as a good example 😎

File: docs/examples/others/test_drag.html [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    //     'chr1',
+    //     0,
+    //     89758810
+    //   );
+    // }, 11000);
+
+
+</script>
+<script>
+  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
```
Comment (flekschas) [inline docs/examples/others/test_drag.html]: Can we remove the GA code from examples?

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "higlass",
-  "version": "1.5.1",
+  "version": "1.5.2",
```
Comment (flekschas) [inline package.json]: Shouldn't this be v1.5.3?

Comment (pkerpedjiev) [reply package.json]: That will sort itself out when merged with master :-)

**Comment by pkerpedjiev** [comment]
> Just minor stuff but what I don't get in general is why does the user need to tell HiGlass first where he/she can drop specific datasets? If I, as a user, have to specify that first then I already know where to drop stuff and don't need visual feedback.

> I guess what I am asking is: can't the track position <-> data type be predefined so I, as a user, don't have to explicitly call the API first?

Good point. I changed the implementation so that you only have to specify the default tracks and the positions will automatically be inferred:

```
{
          server: 'http://higlass.io/api/v1',
          tilesetUid: 'WtBJUYawQzS9M2WVIIHnlA',
          datatype: 'linear-labels',
          defaultTracks: ['linear-labels'],
        }
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,7 @@
+## v1.5.3
```
[the remark this answers is not in the document]
Comment (pkerpedjiev) [reply CHANGELOG.md]: no, I just merged with a develop branch that wasn't up to date so it's missing that part of the CHANGELOG. That will be sorted out when it's merged back.

File: docs/examples/others/test_drag.html (hunk 2 above)
Comment (pkerpedjiev) [reply docs/examples/others/test_drag.html]: Yup, done!

**Pull request merged**